webapp-django/crashstats/crashstats/views.py

```diff
@@ -1103,28 +1105,97 @@ def graphics_report(request):
     if not form.is_valid():
         return http.HttpResponseBadRequest(str(form.errors))
 
-    api = models.GraphicsReport()
-    data = api.get(
-        product=form.cleaned_data['product'] or settings.DEFAULT_PRODUCT,
-        date=form.cleaned_data['date']
-    )
+    batch_size = 1000
+    product = form.cleaned_data['product'] or settings.DEFAULT_PRODUCT
+    date = form.cleaned_data['date']
+    params = {
+        'product': product,
+        'date': [
+            '>={}'.format(date.strftime('%Y-%m-%d')),
+            '<{}'.format(
+                (date + datetime.timedelta(days=1)).strftime('%Y-%m-%d')
+            )
+        ],
+        '_columns': (
+            'signature',
+            'uuid',
+            'date',
+            'product',
+            'version',
+            'build_id',
+            'platform',
+            'platform_version',
+            'cpu_name',
+            'cpu_info',
+            'address',
+            'uptime',
+            'topmost_filenames',
+            'reason',
+            'app_notes',
+            'release_channel',
+        ),
+        '_results_number': batch_size,
+        '_results_offset': 0,
+    }
+    api = SuperSearch()
+    # Do the first query. That'll give us the total and the first page's
+    # worth of crashes.
+    data = api.get(**params)
     assert 'hits' in data
 
     accept_gzip = 'gzip' in request.META.get('HTTP_ACCEPT_ENCODING', '')
     response = http.HttpResponse(content_type='text/csv')
     out = BytesIO()
     writer = utils.UnicodeWriter(out, delimiter='\t')
     writer.writerow(GRAPHICS_REPORT_HEADER)
-    for row in data['hits']:
-        # Each row is a dict, we want to turn it into a list of
-        # exact order as the `header` tuple above.
-        # However, because the csv writer module doesn't "understand"
-        # python's None, we'll replace those with '' to make the
-        # CSV not have the word 'None' where the data is None.
-        writer.writerow([
-            row[x] is not None and row[x] or ''
-            for x in GRAPHICS_REPORT_HEADER
-        ])
+    pages = data['total'] // batch_size
+    pages += data['total'] % batch_size and 1 or 0
```

Comment:
Given that I stopped and stared at this for a while to figure out what it's doing, I claim it's not readable. Seems like it's trying to figure out the total number of pages and to do that it's doing an integer division by `batch_size` and then if there are additional results, adding an additional page. Is that right?

There's got to be a more readable way of expressing that. Maybe something like this?:

```
pages = data['total'] // batch_size
if data['total'] % batch_size:
    pages += 1
```

Maybe combine that with `divmod`?:

```
pages, remainder = divmod(data['total'], batch_size)
if remainder:
    pages += 1
```